src/paas_charm/charm.py

```diff
@@ -498,7 +492,7 @@ def _create_charm_state(self) -> CharmState:
             s3_connection_info=self._s3.get_s3_connection_info() if self._s3 else None,
             saml_relation_data=saml_relation_data,
             rabbitmq_uri=self._rabbitmq.rabbitmq_uri() if self._rabbitmq else None,
-            tracing_relation_data=tracing_relation_data,
+            tempo_parameters=TempoParameters.from_charm(name=self.app.name, tracing=self._tracing),
```

Comment:
I feel like the approach is incoherent compared to how other optional relations are treated. S3 and SAML for example don't use a `from_charm` passed to the charmstate.
Instead they're passing their parameter to the `CharmState` and then it leverages the `IntegrationState.build` to generate the `Parameters` objects.

So what I suggest:
- Check if the relation exists is done in this method for other relation
- Similarly to `saml_relation_data`, build a dict that is either `None` or contains the data required for the TempoParameters
- Pass that dict to the `CharmState.from_charm`, that will use the `IntegrationState.build` with these parameters, to return the `TempoParameters`